### ISSUE TITLE
Add a central location for open pick-up slots

### ISSUE THREAD

**Issue by djahnie** [yunity/karrot-frontend#741]
Sylvia from Groß Gerau complained about the fact, that the new release doesn't show all open slots in pick-ups in a central place anymore. Right now we only show _completely_ empty pick-ups (orange button) and those the currentl logged-in user has already signed up for (green button).

![image](https://user-images.githubusercontent.com/17573771/32725505-a21a70ba-c875-11e7-9665-1440c8822487.png)

How about adding a third (maybe blue..?) button for pick-ups which are not empty but still have open slots?

Original mail from Sylvia:

> Hallo Janina,
> heute ist das neue karrot Release 3 online gegangen.
> Wir von Essen für Alle (EfA) vermissen die Übersicht, bei der alle Betriebe im Kalender zu sehen waren.
> Wo könne wir das finden?
> Wir haben derzeit 11 registrierte Betriebe und 17 Abholer.
> Bei den neuen Release sehe ich aber zunächst nur die Betriebe in denen ich aktuell zur Abholung registriert bin.
> Jeder kann bei EfA in jedem Betrieb abholen, oft kommen einzelne Tage als Ersatzabholung für die Tafel hinzu.
> Es ist also wichtig, dass jeder alle offenen Slots in den Betrieben sieht, damit er schnell reagieren kann.
> Es ist sehr mühsam ggf. mehrfach am Tag alle Betriebe anzuklicken, um zu sehen wie viele Slots besetzt sind und durch wen.
> Liebe Grüße
> Sylvia

**Comment by tiltec** [comment]
I understand that they are asking for the same feature that we had in the angular version, a list of all pickups of all stores in the group.
I think it could make sense to create such a page, but separate from the wall. It could be called "All pickups" and just be a list by date.

I'm unsure where to put it in the mobile navigation, the top navbar is already quite full. Maybe in the "more options" menu of the group? But that's quite hidden...

**Comment by djahnie** [comment]
I think it's important to have it very prominent. And I actually liked to find the pick-up info directly on the main group page, even though I agree that wall and pick-up info are not the same. So maybe it makes sense to seperate them.

How about making one 'pick-ups' page, which displays all the categories of interesting pick-ups and the navbar on mobile shows (from left to right): wall, pick-ups, description - and history moves under the three dots..?

**Comment by tiltec** [comment]
I think we need a plausible concept of a "group homepage" and what it should contain. It seems to me that adding more pickup information into the wall will make the messaging feature less visible and therefore less useful.

I could imagine adding a pickup-centric page as the new group homepage, which shows mainly pickups and only shows messages if there are unread ones (thanks to @nicksellen who made the backend for this in the last months). 

Maybe we should rename this issue into "rethink group homepage"? Or discuss this in #610?

**Comment by djahnie** [comment]
Even though we already have a central pickups page on the dev page, I thought adding the feedback from Bruno could make sense. Yesterday he wrote the following on Slack:

> hey yall nice ppl! I got some feedback from users in Gothenburg on the new version: many people liked the new design and many people complained about one thing: that upcoming pickups do not show up on the first page (because even when 1 out of 3 slots is filled, the pickup disappears from the main page). Is this feature intentional? It makes sense when all slots are filled, but not this way though.

**Comment by tiltec** [comment]
Oh, and this issue is still open! How fitting :)

I'd maybe shift the focus onto the new "pickups" page, reducing the "empty pickups notice" on the wall to maybe 24 hours in advance.

**Comment by djahnie** [comment]
So does 'shifting the focus onto the new pickups page' mean making it the landing page for a group? Or how do you imagine doing it?

(I guess this issue could still be useful to discuss the general idea of a group landing page, like you already suggested above, @tiltec)

**Comment by tiltec** [comment]
> So does 'shifting the focus onto the new pickups page' mean making it the landing page for a group? Or how do you imagine doing it?

I just meant that we don't display so many pickups on the wall anymore. Only the one the user joined and the urgent ones (empty and upcoming in less than 24 hours)

I'm undecided about the other topics, will wait for more feedback.

**Comment by nicco11a** [comment]
I logged a similar issue #791 and was asked what I think about the ideas discussed here so here it comes.. ;)

It's important that a foodsaver directly sees what's available and the current system is confusing because it tells the uses I've joined all the empty pickups. They will not realize there are pickups that need more people, they will not click around anywhere else, just logout because there time is limited so they will not have time to investigate and even if there were a special pickup page most people would miss to click there.

It feels natural to get this information as soon as you login and since you have to expand in order to see empty & non full pickups the point raised about the chat seems to be a none problem since people already seen messages there before they expanded so we are just talking about a longer list. There could be another smaller headline in the list for non full ones to make it clear. 

I am not 100% sure why the distinction, I would just call it available pickups. I can guess that the idea is that it's more important that empty pickups are booked rather then all slots. I am feeling like we at Solikyl prefer pickups to be fully booked rather then just one person who will have to do all the work then.

**Comment by tiltec** [comment]
Some proposals I gathered:

1. make the "all pickups in this group" page the default group page
2. add another bar below "joined pickups" and "empty pickups" that could be called "available pickups"
3. re-purpose "empty pickups" to show also non-empty pickups (but not full pickups)

What do you all think of these?

Slightly related: the group conversation that's now only available on the wall could get it's own section and we add a "Dashboard" for the summaries (or keep the name "Wall"). But that's a bit too big for this discussion.

I'm a bit short on laptop time right now (still no grid power in this building where I'm in), so I might not implement these changes in the next days, but I'm happy about other proposals and discussion.

**Comment by brnsolikyl** [comment]
I'm in favor of solution 3: just show all coming pickups that are either empty or available. The tab can be renamed to available pickups since it also encompasses empty pickups.

However, any of the solutions would be good now for us right now, because I noticed a drop in the number of people joining pickups, even though we have increased in number of registered users while maintaining the number of pickups/slots. I guess people are used to the older version and cannot easily find available pickups.
